### PR TITLE
chore: don't hard-code cdk build order

### PR DESCRIPTION
**Pull request by jelbourn**
@DevVersion will using grep like this be a problem for Windows?

### CODE REVIEW

File: tools/package-tools/secondary-entry-points.ts [hunk 1]
```diff
@@ -1,35 +1,82 @@
...
  *
  * @param packageName The package name for which to get entry points, e.g., 'cdk'.
  * @returns An array of secondary entry-points names, e.g., ['a11y', 'bidi', ...]
  */
 export function getSecondaryEntryPointsForPackage(packageName: string) {
-  // For now, we hard-code the fact that only the CDK has secondary entry-points until we switch
-  // to bazel.
-  if (packageName === 'cdk') {
-    return CDK_ENTRY_POINTS;
+  const packageDir = join(buildConfig.packagesDir, packageName);
```
Comment (jelbourn) [inline tools/package-tools/secondary-entry-points.ts]: Good point; now uses the `BuildPackage`

File: tools/package-tools/secondary-entry-points.ts [hunk 2]
```diff
@@ -1,35 +1,82 @@
...
+
+  // First get the list of all entry-points as the list of directories in the package excluding
+  // blacklisted directories (e.g., "testing").
+  const entryPoints = readdirSync(packageDir)
+      .filter(f => lstatSync(join(packageDir, f)).isDirectory() && DIR_BLACKLIST.indexOf(f) < 0);
```
Comment (jelbourn) [inline tools/package-tools/secondary-entry-points.ts]: Had to add `es2016.array.include` to the tsconfig in `tools/gulp`.

File: tools/package-tools/secondary-entry-points.ts [hunk 3]
```diff
@@ -1,35 +1,83 @@
...
+
+  // Update the deps for each node to point to the appropriate BuildNodes.
+  buildNodes.forEach(node => {
+    // Look for any imports that reference this same umbrella package and get the corresponding
+    // BuildNode for each by looking at the import statements with grep.
+    node.deps = spawnSync('egrep', [
+      '-roh',
+      '--include', '*.ts',
+      `from.'@angular/${packageName}/.+';`,
+      `./src/${basename(packageDir)}/${node.name}/`
```
Comment (DevVersion) [inline tools/package-tools/secondary-entry-points.ts]: Why not just `packageRoot` here?

**Comment by jelbourn** [comment]
Updated

**Pull request merged**